modules/tensor_mechanics/doc/content/source/materials/ADRadialReturnStressUpdate.md

```diff
@@ -104,6 +104,12 @@ class.
 More details on how to write the equivalent yield surface equation for a creep model are given in
 Dunne and Petrinic.
 
+## Substepping
+
+We provide the substepping capability in `ADRadialReturnStressUpdate` for nonlinear material models in order to improve the convergence. The idea is that when material is undergoing large deformation and the return mapping algorithm struggles to converge, we would divide the original strain into smaller strain increments and take several substeps where incremental strain is applied at each substep.  The following shows an example of the syntax for using substepping.
```

Comment:
I thought the idea was to be able to use substepping to maximize global timestep size while minimizing maximum strain increments by all other things besides stress are constant during creep rate evaluations (i.e. temperature, porosity, elasticity tensor, etc)?

Reply:
My understanding is that sub stepping could help convergence with large global timestep under the cost of many substeps and internal nonlinear loops (return mapping solves). Ideally this would be triggered when a return mapping solve fails, which indicates high nonlinearity of the problem. This initial implementation, which calculates the substep based on the maximum allowed strain, would ideally end up with multiple substep for the 'problematic' quadrature points and only one substep for the rest. For cases with many substeps, the strain, stress, and internal state variables are still evolving via `computeStressInitialize`, `computeStressFinalize`, etc in the specific material model.